src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,82 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object classImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        checkAnnotation(clazz);
+        Field[] fieldsDeclaration = clazz.getDeclaredFields();
```

Comment:
Maybe better to rename variable **fieldsDeclaration** to **declaredFields** or **fields**. **fieldsDeclaration** is confusing because we don't declare fields in this row, we get it and try to initialize them below, they are already declared in classes